`BreakLoopOp` does not have `blocks` attribute expected by pass `ContractIdleWiresInControlFlow`

### Environment

- **Qiskit version**: 2.3.0
- **Python version**: 3.14.0
- **Operating system**: Arch Linux


### What is happening?

Calling `transpile()` on a circuit with break_loop ops fails because the pass `ContractIdleWiresInControlFlow` calls `_contract_control_flow()`, which assumes that all ops have the attribute `blocks`.

This issue causes the test `test/terra/backends/aer_simulator/test_control_flow.py::TestControlFlow::test_transpile_break_and_continue_loop` from qiskit_aer to fail. See https://github.com/Qiskit/qiskit-aer/issues/2354

### How can we reproduce the issue?

1. Install qiskit 2.3.0
2. Get tests from qiskit_aer
3. Run e.g. `python -m pytest test/terra/backends/aer_simulator/test_control_flow.py`

### What should happen?

Test should not fail

### Any suggestions?

I guess the easiest workaround would be to check if the op object has attribute `blocks` before line 60:

https://github.com/Qiskit/qiskit/blob/9186f49af0396d82e7d9880602e719a46826ba1d/qiskit/transpiler/passes/optimization/contract_idle_wires_in_control_flow.py#L60C18-L60C27

Something like this:

```python
if not hasattr(op, "blocks"):
   return inst
```

Is the PR mentioned above supposed to fixed the issue by itself? Or should it be apply together with some other commit. Because if it's the first, it does not solve the issue running the Qiskit Aer unit tests.

This is the error:

```
dag_mapping = <function trivial_recurse.<locals>.out.<locals>.bound_wrapped_method at 0x7f1d3c435850>
op = Instruction(name='break_loop', num_qubits=1, num_clbits=1, params=[])

    def map_blocks(dag_mapping: Callable[[DAGCircuit], DAGCircuit], op: ControlFlowOp) -> ControlFlowOp:
        """Use the ``dag_mapping`` function to replace the blocks of a :class:`.ControlFlowOp` with new
        ones.  Each block will be automatically converted to a :class:`.DAGCircuit` and then returned
        to a :class:`.QuantumCircuit`."""
>       return op.replace_blocks(
               ^^^^^^^^^^^^^^^^^
            [
                dag_to_circuit(dag_mapping(circuit_to_dag(block)), copy_operations=False)
                for block in op.blocks
            ]
        )
E       AttributeError: 'BreakLoopOp' object has no attribute 'replace_blocks'
```

--- a/crates/circuit/src/dag_circuit.rs
+++ b/crates/circuit/src/dag_circuit.rs
@@ -3754,7 +3754,12 @@ impl DAGCircuit {
             .node_references()
             .filter_map(|(node_index, node_type)| match node_type {
                 NodeType::Operation(node) => {
-                    if node.op.try_control_flow().is_some() {
+                    if node.op.try_control_flow().is_some_and(|control_flow| {
+                        !matches!(
+                            control_flow.control_flow,
+                            ControlFlow::BreakLoop | ControlFlow::ContinueLoop
+                        )
+                    }) {
                         Some(self.unpack_into(py, node_index, node_type))
                     } else {
                         None

--- a/test/python/circuit/test_control_flow.py
+++ b/test/python/circuit/test_control_flow.py
@@ -18,6 +18,7 @@ import math
 from ddt import ddt, data, unpack, idata
 
 from qiskit.circuit import Clbit, ClassicalRegister, Instruction, Parameter, QuantumCircuit, Qubit
+from qiskit import transpile
 from qiskit.circuit.classical import expr, types
 from qiskit.circuit.controlflow import CASE_DEFAULT, condition_resources, node_resources
 from qiskit.circuit.library import XGate, RXGate
@@ -1281,3 +1282,17 @@ class TestAddingControlFlowOperations(QiskitTestCase):
             )
         with self.assertRaisesRegex(CircuitError, "not in this circuit"):
             outer.append(BoxOp(inner.copy()), [0], [0])
+
+    def test_transpiling_with_control_flow(self):
+        """Test we don't run into compilation errors when transpiling circuits with control flows"""
+        qc = QuantumCircuit(1, 1)
+
+        with qc.for_loop(range(1000)):
+            qc.h(0)
+            qc.measure(0, 0)
+            with qc.if_test((0, False)):
+                qc.continue_loop()
+            qc.break_loop()
+
+        transpiled = transpile(qc)
+        self.assertIsNotNone(transpiled)

